internal/inventory/asset.go

```diff
@@ -329,9 +329,32 @@ type AssetResourcePolicy struct {
 	Condition  map[string]any `json:"condition,omitempty"`
 }
 
+// EntityMetadata maps metadata required to use Entity Store
+type entityMetadata struct {
+	Category string `json:"category"`
+	Type     string `json:"type"`
+}
+
 // AssetEnricher functional builder function
 type AssetEnricher func(asset *AssetEvent)
 
+func (a *AssetEvent) getEntityMetadata() map[string]entityMetadata {
+	ids := a.Asset.Id
+
+	if len(ids) == 0 {
+		return nil
+	}
+
+	// Picking up only first id, we need to make a decision on if we
+	// have a "primary" id or if we duplicate data
```

Comment:
That's super important. We also have to pay attention to always put "the" asset ID in the `ids[0]` spot.

Reply:
This is a product question I want to clear it up. I believe it was a bad decision in the end to have multiple ids.

We need to have a standardized id per resource. We need to make the decision for example is aws iam user arn or username? And one for each one of those.

The initial attempt of having a "correlation best chance" approach isn't going to help us in the asset inventory, unfortunately.